### PR TITLE
Swaparoo Contract - Added text explainer to go with video

### LINKED ISSUE #940: Page 2 - How to Swap assets
Goal: Use swaparoo contract and modify it for "seatgeek" use-case.
https://github.com/agoric-labs/dapp-swaparoo

Involves:

- Invitation specs
- Offer shapes

- Depends on Client docs #922 

Create a (Lesson plan + video) that shows the code and explainer for the following:

- How to Swap assets (Swaparoo) | with Code sample

### PR DESCRIPTION
**Pull request by kbennett2000**
closes #940

### CODE REVIEW

File: main/guides/getting-started/swaparoo-how-to-swap-assets-explainer.md [hunk 1]
```diff
@@ -1,5 +1,106 @@
 # Swaparoo Contract
 
-## How to Swap Assets
+This smart contract is designed to allow two parties to swap assets between themselves, with a fee charged to one of the parties. The contract is started with a `feeAmount` and a `namesByAddressAdmin` object, which is used to retrieve a deposit facet for the second party.
```
Comment (dckc) [inline main/guides/getting-started/swaparoo-how-to-swap-assets-explainer.md:3]: `namesByAddressAdmin` is actually excess authority; read-only access, i.e. `namesByAddress` should be enough. But it's not, due to a bug:

 - https://github.com/Agoric/agoric-sdk/issues/8113

the bug is fixed in master and due to hit mainnet in upgrade-14, I think. 

dunno how important it is to mention this.

File: main/guides/getting-started/swaparoo-how-to-swap-assets-explainer.md [hunk 2]
```diff
@@ -1,5 +1,106 @@
...
+
+Let's take a look at how this contract works:
+
+## Setting up Fee Handling
+The contract retrieves the `feeIssuer` from the Zoe service, which is an issuer for the stable coin used for fees. It creates a `feeSeat` and a `feeShape` based on the `feeAmount` specified in the contract terms.
```
Comment (dckc) [inline main/guides/getting-started/swaparoo-how-to-swap-assets-explainer.md:8]:
```suggestion
The contract retrieves the `feeIssuer` from the Zoe service, which is an issuer for the stable token used for fees. It creates a `feeSeat` and a `feeShape` based on the `feeAmount` specified in the contract terms.
```

**important**
cf. https://agoric.com/blog/getting-started/ist

File: main/guides/getting-started/swaparoo-how-to-swap-assets-explainer.md [hunk 3]
```diff
@@ -1,5 +1,106 @@
...
+const stableIssuer = await E(zcf.getZoeService()).getFeeIssuer();
+const feeBrand = await E(stableIssuer).getBrand();
+const { zcfSeat: feeSeat } = zcf.makeEmptySeatKit();
+const feeShape = makeNatAmountShape(feeBrand, feeAmount.value);
+```
+
+## Making the First Invitation
+The `makeFirstInvitation` function is called with an array of issuers. It verifies that these issuers are part of the contract terms and saves any new issuers to the contract. It then creates an invitation with a proposal shape that includes the `feeShape` in the give record.
+```js
+const makeFirstInvitation = issuers => {
```
Comment (dckc) [inline main/guides/getting-started/swaparoo-how-to-swap-assets-explainer.md:19]: _just a thought..._

I wonder how to keep these code excerpts in sync as dapp-agoric-basics changes.

The whole zoe section is wildly out of date due to using game terminology when dapp-offer-up has moved on :-/

**Review by dckc — CHANGES_REQUESTED** [review]
one important change

also, for approval, I'm inclined to defer to @sufyaankhan

**Comment by dckc** [comment]
oh... also... do we want to keep https://docs.agoric.com/guides/zoe/contracts/atomic-swap.html ? if so, we should have links in both directions

**Comment by kbennett2000** [comment]
Closes #940

**Comment by kbennett2000** [comment]
> oh... also... do we want to keep https://docs.agoric.com/guides/zoe/contracts/atomic-swap.html ? if so, we should have links in both directions

I'll defer to @sufyaankhan on that one...

**Comment by dckc** [comment]
> Closes #940

I think it has to go in the description... let's see...

> in the pull request's description or in a commit message
 -- [Linking a pull request to an issue using a keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

File: main/guides/getting-started/swaparoo-how-to-swap-assets-explainer.md (hunk 3 above)
Comment (kbennett2000) [reply main/guides/getting-started/swaparoo-how-to-swap-assets-explainer.md:19]: Would a Github workflow in the app repos that create an issue in the documentation repo when something changes work?

File: main/guides/getting-started/swaparoo-how-to-swap-assets-explainer.md (hunk 1 above)
Comment (kbennett2000) [reply main/guides/getting-started/swaparoo-how-to-swap-assets-explainer.md:3]: Added a note about it

**Comment by kbennett2000** [comment]
> since this is a new page, it needs to be referenced in nav.js and/or config.mjs

@sufyaankhan - did we decide where in the navigation these explainers are going to go?

**Pull request closed without merging**